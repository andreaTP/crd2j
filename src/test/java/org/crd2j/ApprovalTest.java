package org.crd2j;

import io.fabric8.kubernetes.client.DefaultKubernetesClient;
import io.fabric8.kubernetes.client.KubernetesClient;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.util.List;
import java.util.Optional;

import static org.assertj.core.api.Assertions.assertThat;

public class ApprovalTest {

    KubernetesClient client = new DefaultKubernetesClient();
    CRGeneratorRunner runner = new CRGeneratorRunner();

    @Test
    void testCrontabCrd() {
        // Arrange
        var crd = client
                .apiextensions()
                .v1()
                .customResourceDefinitions()
                .load(this
                        .getClass()
                        .getClassLoader()
                        .getResource("crontab-crd.yml"))
                .get();

        // Act
        var writables = runner.generate(crd, Optional.empty());

        // Assert
        assertThat(writables.size()).isEqualTo(1);

        var writable = writables.get(0);

        Approvals.verifyAll("CrontabJavaCr", List.of(
            writable.getJavaClass("CronTab"),
            writable.getJavaClass("Spec"),
            writable.getJavaClass("Status"))
        );
    }

    @Test
    void testKeycloakCrd() {
        // Arrange
        var crd = client
                .apiextensions()
                .v1()
                .customResourceDefinitions()
                .load(this
                        .getClass()
                        .getClassLoader()
                        .getResource("keycloak-crd.yml"))
                .get();

        // Act
        var writables = runner.generate(crd, Optional.empty());

        // Assert
        assertThat(writables.size()).isEqualTo(1);

        var writable = writables.get(0);

        Approvals.verifyAll("KeycloakJavaCr", List.of(
                writable.getJavaClass("Keycloak"),
                writable.getJavaClass("Spec"),
                writable.getJavaClass("Status"))
        );
    }

    @Test
    void testAkkaMicroservicesCrd() {
        // Arrange
        var crd = client
                .apiextensions()
                .v1()
                .customResourceDefinitions()
                .load(this
                        .getClass()
                        .getClassLoader()
                        .getResource("akka-microservices-crd.yml"))
                .get();

        // Act
        var writables = runner.generate(crd, Optional.empty());

        // Assert
        assertThat(writables.size()).isEqualTo(1);

        var writable = writables.get(0);

        Approvals.verifyAll("AkkaMicroservicesJavaCr", List.of(
                writable.getJavaClass("AkkaMicroservice"),
                writable.getJavaClass("Spec"),
                writable.getJavaClass("Status"))
        );
    }
}
