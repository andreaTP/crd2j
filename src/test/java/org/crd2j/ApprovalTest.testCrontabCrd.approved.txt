CrontabJavaCr[0] = package v1;

@io.fabric8.kubernetes.model.annotation.Version("stable.example.com")
@io.fabric8.kubernetes.model.annotation.Group("v1")
public class CronTab extends io.fabric8.kubernetes.client.CustomResource<Spec, Status> implements io.fabric8.kubernetes.api.model.Namespaced {
}
CrontabJavaCr[1] = package v1;

public class Spec {

    private String image;

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    private String cronSpec;

    public String getCronSpec() {
        return cronSpec;
    }

    public void setCronSpec(String cronSpec) {
        this.cronSpec = cronSpec;
    }

    private Long replicas;

    public Long getReplicas() {
        return replicas;
    }

    public void setReplicas(Long replicas) {
        this.replicas = replicas;
    }
}
CrontabJavaCr[2] = package v1;

public class Status {

    private Long replicas;

    public Long getReplicas() {
        return replicas;
    }

    public void setReplicas(Long replicas) {
        this.replicas = replicas;
    }

    private String labelSelector;

    public String getLabelSelector() {
        return labelSelector;
    }

    public void setLabelSelector(String labelSelector) {
        this.labelSelector = labelSelector;
    }
}
